options.h

```diff
@@ -293,6 +293,7 @@ struct pattern_options {
 	uint8_t verifier_feed_zone_is_default;
 	int32_t verifier_timeout;
 	uint8_t verifier_timeout_is_default;
+	const char* catalog;
```

Comment:
This seems off(?) I'd expect this to be a boolean value(?) Either a zone is a catalog zone or not? Based on whether the server is a primary or secondary, that'd have different implications. Currently it almost seems to specify which catalog to use, but that behavior is implemented by `include-pattern`. The pattern itself would dictate where to get the catalog zone from? The boolean value would merely indicate post-processing is necessary. Then when reading the catalog zone file, we'd mark other zones as necessary automatically?